src/QMCWaveFunctions/OptimizableObject.h

```diff
@@ -0,0 +1,67 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_OPTIMIZABLEOBJECT_H
+#define QMCPLUSPLUS_OPTIMIZABLEOBJECT_H
+
+#include "VariableSet.h"
+
+/**@file OptimizableObject.h
+ *@brief Declaration of OptimizableObject
+ */
+namespace qmcplusplus
+{
+using opt_variables_type = optimize::VariableSet;
+
+class OptimizableObject
+{
+public:
+  OptimizableObject(const std::string& name, bool optimizable) : name_(name), optimizable_(optimizable) {}
+
+  const std::string& getName() const { return name_; }
+  bool isOptimized() const { return is_optimized_; }
+  bool isOptimizable() const { return optimizable_; }
+
+private:
+  /** Name of the optimizable object
+   */
+  const std::string name_;
+  /** If true, this object and/or its underlying components is optimizable_
+   */
+  const bool optimizable_;
+  /** If true, this object is actively modified during WFOpt
+   */
+  bool is_optimized_ = false;
+
+public:
+  /** check in variational parameters to the global list of parameters used by the optimizer.
+   * @param active a super set of optimizable variables
+   *
+   * This is a query function and should never be implemented as a feature blocker.
+   * If an SPOSet derived class doesn't support optimization, use the base class fallback.
```

Comment:
Mention of "SPOSet".  This should probably be OptimizableObject?

Reply:
But the fall back is to do nothing.  Which means that by default classes deriving from OptimizableObject aren't optimizable.  I really think it would be better to get a vector of optimization inteface references belonging to actual optimizable objects and have that be the API.

A vector of "OptimizableObjects" that also includes many objects that cannot be optimized but have a default implementation of noop for optimization is less desirable.

Reply:
Good catch.